cocotb/decorators.py

```diff
@@ -291,6 +287,7 @@ class coroutine(object):
     """
 
     def __init__(self, func):
+        func = types.coroutine(func)
```

Comment:
As best I can tell, the only benefit to this line is that:
* `yield from some_async_function` would work inside cocotb coroutines
  * No one is using `yield from` in cocotb today anyway, so this provides little value. I suppose what it does do is allow users to convert to async functions from the bottom of the stack out and gain a tiny bit of performance over using `yield some_async_function`, but I don't think confusing users with multiple spellings is worth it
* `await some_coro._func(...)` would work inside async functions.
  * This attribute is private.